src/containers/shared/components/Streams.jsx

```diff
@@ -2,30 +2,69 @@ import { Component } from 'react';
 import PropTypes from 'prop-types';
 import Log from '../log';
 import { fetchNegativeUNL, fetchQuorum, fetchMetrics } from '../utils';
-import {
-  handleValidation,
-  handleLedger,
-  fetchLedger,
-  fetchLoadFee,
-} from '../../../rippled/lib/streams';
 import SocketContext from '../SocketContext';
+import { getLedger, getServerInfo } from '../../../rippled/lib/rippled';
+import { summarizeLedger, EPOCH_OFFSET } from '../../../rippled/lib/utils';
 
 const MAX_LEDGER_COUNT = 20;
 
+const PURGE_INTERVAL = 10 * 1000;
+const MAX_AGE = 120 * 1000;
+
 const throttle = (func, limit) => {
   let inThrottle;
+  let queued = false;
   return function throttled(...args) {
     const context = this;
     if (!inThrottle) {
+      queued = false;
       func.apply(context, args);
       inThrottle = true;
       setTimeout(() => {
+        if (queued) {
```

Comment:
Does this check the outside state `queued`, or does this reference the value of `queued` when the timeout was started?

Reply:
It looks at the current value of queue.  So subsequent calls to the wrapped throttle function will set queue to true if the happen during throttling.